code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -56,8 +61,48 @@
 						return
 					M.put_in_l_hand(src)
 			add_fingerprint(usr)
-			return
 
+//Proc for Contents transfer to other containers, bins, turf, and more possible additions.
+/obj/item/weapon/storage/proc/contentto(obj/dest_object, mob/user)
+	playsound(loc, "rustle", 50, 1, -5)
+	switch( storage_contents_dump_act(dest_object) )

```

Comment:
I must apologise but I find this a <I>hilarious</I> misinterpretation of my suggestions/intent.

storage_contents_dump_act() would CONTAIN the type specific code, instead of returning a value that then points to a switch statement containing the type specific code.

you'd call storage_contents_dump_act(dest_object) and that'd be it, all the type specific code is in the children procs for those types.